### PR TITLE
Re-enable checks for resource allocation failure

### PR DESCRIPTION
**Pull request by gordonwang0**

### CODE REVIEW

File: libraries/standard/mqtt/src/iot_mqtt_api.c [hunk 1]
```diff
@@ -469,8 +469,8 @@ static bool _createKeepAliveOperation( const IotMqttNetworkInfo_t * pNetworkInfo
     pMqttConnection->pingreq.u.operation.periodic.ping.nextPeriodMs = ( uint32_t ) keepAliveSeconds * 1000U;
 
     /* Generate a PINGREQ packet. */
-    serializeStatus = _getMqttPingreqSerializer( pMqttConnection->pSerializer )( &( pMqttConnection->pingreq.u.operation.pMqttPacket ),
-                                                                                 &( pMqttConnection->pingreq.u.operation.packetSize ) );
+    serializeStatus = _getMqttPingreqSerializer( pNetworkInfo->pMqttSerializer )( &( pMqttConnection->pingreq.u.operation.pMqttPacket ),
```
Comment (muneebahmed10) [inline libraries/standard/mqtt/src/iot_mqtt_api.c]: Wondering why some of these in this file use `pNetworkInfo->pMqttSerializer` while others use `pMqttConnection->pSerializer`

Comment (gordonwang0) [reply libraries/standard/mqtt/src/iot_mqtt_api.c]: This was a bug.  This is during the initialization of the MQTT connection, before the MQTT connection has been set up.  Therefore, the serializer needs to be read from the parameter, pNetworkInfo.

File: libraries/standard/mqtt/include/types/iot_mqtt_types.h [hunk 2]
```diff
@@ -797,7 +797,7 @@ typedef IotMqttError_t ( * IotMqttSerializeSubscribe_t )( const IotMqttSubscript
  * @param[out] uint8_t** Where the DISCONNECT packet is written.
  * @param[out] size_t* Size of the DISCONNECT packet.
  */
-typedef IotMqttError_t ( * IotMqttSerializeDisconnect_t )( uint8_t ** ppDisconnectPacket,
+typedef IotMqttError_t ( * IotMqttSerializeDisconnect_t )( uint8_t ** pDisconnectPacket,
```
Comment (dcgaws) [inline libraries/standard/mqtt/include/types/iot_mqtt_types.h]: We're using "p" Hungarian, but not "pp"?

File: libraries/standard/mqtt/test/unit/iot_tests_mqtt_api.c [hunk 3]
```diff
@@ -999,6 +992,9 @@ TEST( MQTT_Unit_API, ConnectMallocFail )
     int32_t i = 0;
     IotMqttError_t status = IOT_MQTT_STATUS_PENDING;
     IotMqttConnectInfo_t connectInfo = IOT_MQTT_CONNECT_INFO_INITIALIZER;
+    IotMqttSerializer_t serializer = IOT_MQTT_SERIALIZER_INITIALIZER;
+
+    _pAllocatedPingreq = NULL;
```
Comment (dcgaws) [inline libraries/standard/mqtt/test/unit/iot_tests_mqtt_api.c]: Why does this need to be re-initialized?

File: libraries/standard/mqtt/test/unit/iot_tests_mqtt_api.c [hunk 4]
```diff
@@ -165,6 +165,11 @@ static IotMqttNetworkInfo_t _networkInfo = IOT_MQTT_NETWORK_INFO_INITIALIZER;
  */
 static IotNetworkInterface_t _networkInterface = { 0 };
 
+/**
+ * @brief A packet allocated by _serializePingreq.
+ */
+static uint8_t * _pAllocatedPingreq = NULL;
```
Comment (dcgaws) [inline libraries/standard/mqtt/test/unit/iot_tests_mqtt_api.c]: Why does this need to be global?

File: libraries/standard/mqtt/include/types/iot_mqtt_types.h (hunk 2 above)
Comment (gordonwang0) [reply libraries/standard/mqtt/include/types/iot_mqtt_types.h]: Yes.  This change is just for consistency.

File: libraries/standard/mqtt/test/unit/iot_tests_mqtt_api.c (hunk 3 above)
Comment (gordonwang0) [reply libraries/standard/mqtt/test/unit/iot_tests_mqtt_api.c]: It doesn't

File: libraries/standard/mqtt/test/unit/iot_tests_mqtt_api.c (hunk 4 above)
Comment (gordonwang0) [reply libraries/standard/mqtt/test/unit/iot_tests_mqtt_api.c]: It's shared between _serializePingreq and the test function.  It's set by _serializePingreq so the test function knows what to free to clean up.

**Pull request merged**